### PR TITLE
handle negative infinity timeout

### PR DESCRIPTION
**Pull request by SimenB**
See this issue in Jest, with links to spec: https://github.com/facebook/jest/issues/5960. Without changing the implementation of Lolex, Jest test suite breaks when attempting to use it

### CODE REVIEW

File: test/lolex-test.js [hunk 1]
```diff
@@ -280,6 +280,20 @@ describe("lolex", function () {
...
+                calls.push("NaN");
+            }, NaN);
+            this.clock.setTimeout(function () {
+                calls.push("Infinity");
+            }, Number.POSITIVE_INFINITY);
+            this.clock.setTimeout(function () {
+                calls.push("-Infinity");
+            }, Number.NEGATIVE_INFINITY);
+            this.clock.runAll();
+            assert.equals(calls, ["NaN", "Infinity", "-Infinity"]);
```
Comment (SimenB) [inline test/lolex-test.js]: without the code change, `-Infinity` is before `Infinity` in the array

File: src/lolex-src.js [hunk 2]
```diff
@@ -196,6 +208,9 @@ function addTimer(clock, timer) {
     timer.type = timer.immediate ? "Immediate" : "Timeout";
 
     if (timer.hasOwnProperty("delay")) {
+        if (!isNumberFinite(timer.delay)) {
```
Comment (benjamingr) [inline src/lolex-src.js]: So I checked - the spec actually says (if < 0 then = 0) but Infinity isn't explicitly treated - >Int32 numbers overflow and also get set to 0 but not explicitly. 

Node on the other hand sets the value to 1 (and not 0) https://github.com/nodejs/node/blob/master/lib/internal/timers.js#L19 done in https://github.com/nodejs/node/blob/master/lib/internal/timers.js#L55

Comment (benjamingr) [reply src/lolex-src.js]: https://github.com/nodejs/node/blob/master/lib/internal/timers.js#L55

Comment (SimenB) [reply src/lolex-src.js]: (I've never read much spec, so all questions are in good faith)

From the linked issue in Jest, I read it as point 8 in https://heycam.github.io/webidl/#abstract-opdef-converttoint (as it's a long, not `EnforceRange`), namely

>If x is NaN, +0, +∞, or −∞, then return +0.

Comment (benjamingr) [reply src/lolex-src.js]: Yeah, exactly.

Comment (SimenB) [reply src/lolex-src.js]: Oh, so we are in agreement. Cool! 😀 

Do you think it's a bug in Node?

Comment (benjamingr) [reply src/lolex-src.js]: @bnoordhois https://github.com/nodejs/node/commit/7fc835afe362ebd30a0dbec81d3360bd24525222 - 

Do you happen to remember why timers are set to 1 and not 0?

Comment (SimenB) [reply src/lolex-src.js]: Seeing as the tag failed, I'll be a bit cheeky and do @bnoordhuis

Comment (fatso83) [reply src/lolex-src.js]: So glad you guys are looking into the sweaty details of this. Delegation is :crown: Hope you add some comments to the implementation to make code archaeology easier for future maintainers :)

Comment (benjamingr) [reply src/lolex-src.js]: I've opened an issue over at core about the bahvior - I think it's fine to be honest, it's not what the timers spec says but it's what Chrome and Node do basically https://github.com/nodejs/node/issues/20596

Comment (SimenB) [reply src/lolex-src.js]: FWIW, I'm fine setting it to `1`. That's not observable from my test - the test is the order in which the callbacks gets invoked, not if the timeout is `0` or `1`.

**Comment by fatso83** [comment]
2.4.1 out.

**Comment by SimenB** [comment]
Awesome. Updated the PR in Jest. Getting closer!

**Pull request merged**